sisl/io/siesta/binaries.py

```diff
@@ -14,10 +14,10 @@
     print(e)
     found_module = False
 
-from ..sile import add_sile, SileError
+from ..sile import add_sile, SileError, SileWarning
```

Comment:
By the way, I caught this one (`SileWarning` was used in a method but not imported) because in VS code I have the Pylance extension which does static code analysis on the files you open. It reinforces the need for #295 to effortlessly make the code cleaner (remove unused imports) and less buggy :)

Reply:
Nice catch, yeah at some point I had an analyser, however it annoyingly didn't catch some things, and too many of others. Definitely we should have a static analyzer!

Reply:
`flake8` seems to be a quite good static code analyzer.

You can test it very quickly. To install:

```
python -m pip install flake8
```

Then to check for undeclared variables (error `F821`):
```
flake8 --select F821 path/to/sisl
```

There are quite a few :)

Reply:
damn ;)

Reply:
FYI, I already have configure flake8, but just never ran it periodically... :(

Reply:
I fixed everything except a `display` call in your viz modules:
```
sisl/viz/plot.py:1314:9: F821 undefined name 'display'
```
I have tried to fix this in 608616c6f1

Reply:
Yes, that works! :+1: